### PR TITLE
Feature/fres 1522 single answer

### PR DESCRIPTION
**Pull request by maxired**
Hi,

this PR implements the fact that a question can have a single answer.
If so, it does not show the answer zone.

For spaces created with the inventory item, using game definition with this new feature requires the space the be edited.

Screenshot 2022-08-18 at 17 55 45

can be tested with the following card:

```csv
card,weight,answer_yes,answer_no
Are you ready ?, 100,YES!,
Do you want to stop ?, 100,,NO!
```

### CODE REVIEW

File: elements/reigns/src/features/game/validateGameDefinition.test.ts [hunk 1]
```diff
@@ -13,176 +13,174 @@ import demo from "../../../public/games/demo.json";
...
-            card: "some card",
-            weight: 1,
-            cooldown: 3.5 as any,
-          } as Card,
-        ])
+        validateCards([createDefaultCard("some card", { cooldown: 3.5 })])
       ).toThrow();
     });
+
+    it("should thrown if no answer_yes and no answer_no", () => {
```
Comment (staff0rd) [inline elements/reigns/src/features/game/validateGameDefinition.test.ts:154]:
```suggestion
    it("should throw if no answer_yes and no answer_no", () => {
```

File: elements/reigns/src/features/game/validateGameDefinition.test.ts [hunk 2]
```diff
@@ -13,176 +13,174 @@ import demo from "../../../public/games/demo.json";
...
+      const card = createDefaultCard();
+      delete (card as any).answer_no;
+      delete (card as any).answer_yes;
+
+      expect(() => validateCards([card])).toThrow();
+    });
+
+    it("should not throw if with a valid answer_yes and no answer_no", () => {
```
Comment (staff0rd) [inline elements/reigns/src/features/game/validateGameDefinition.test.ts:162]:
```suggestion
    it("should not throw when a valid answer_yes and no answer_no", () => {
```

File: elements/reigns/src/features/voting/useVoteListener.ts [hunk 3]
```diff
@@ -2,20 +2,28 @@ import { useEffect } from "react";
...
 
 export const useVoteListener = (phase: GamePhase) => {
   useEffect(() => {
     if (phase === GamePhase.STARTED) {
       const sdk = getSdk();
       const yesListener = sdk.subscribeToGlobalEvent("custom.reign.yes", () => {
-        Logger.log(Logger.VOTE, "vote yes");
-        persistParticipantVote(sdk.localParticipant.id, "Yes");
+        console.log("maxired yes listener", new Game().retrieve().selectedCard);
+        if (new Game().retrieve().selectedCard?.answer_yes) {
```
Comment (staff0rd) [inline elements/reigns/src/features/voting/useVoteListener.ts:14]:
```suggestion
        if (new Game().retrieve().selectedCard?.answer_yes) {
```

**Review by staff0rd — APPROVED** [review]

**Comment by maxired** [comment]
@staff0rd thanks for the review and suggestions !

**Pull request merged**